agent/run.sh

```diff
@@ -140,6 +140,10 @@ launch_agent()
         export no_proxy=$NO_PROXY
     fi
 
+    if [ -n "$CATTLE_CHECK_NAMESERVER" ]; then
```

Comment:
Why do you need this block that checks if the env var is set and then exports it? Presumably, its set because it was passed into the docker run command as `-e CATTLE_CHECK_NAMESERVER=false` and the other line you added below (`-e CATTLE_CHECK_NAMESERVER`) is enough to pass it on to the rancher/agent container.